### PR TITLE
 Fix how `preprocess_array` (openpifpaf/yolact) handles float32/64 input

### PR DESCRIPTION
**Pull request by dbogunowicz**
## Feature Description

`preprocess_array` is an input preprocessing function that is used by both yolact and openpifpaf (yeah, it's pretty messy, we should standardize the preprocessing for all the CV pipelines at some point). 

Back when I wrote it, I only considered that preprocessing function in the context of the annotation i.e. the input is a path to the model -> we will be effectively passing the uint8 array. Because of this explicit assumption `preprocess_array` e.g. always divides the input by 255, which is clearly incorrect if we pass float32 arrays.

This PR rectifies this error i.e. makes the function robust to both float32 and uint8 image inputs.

## Manual Testing
```python
import cv2
from deepsparse import Pipeline
import numpy
input_uint8 = "src/deepsparse/open_pif_paf/sample_images/dancers.jpg"
input_float32 = cv2.imread(input_uint8).astype(numpy.float32) / 255

## Manual test for openpifpaf pipeline
pipeline = Pipeline.create(task="open_pif_paf", model_path="openpifpaf-resnet50.onnx")
output_uint8 = pipeline(images=input_uint8)
output_float32 = pipeline(images=input_float32)
# scores are close but not the same
# (1st case we resize the uint8 and then convert to float32)
# (2nd case we convert the uint8 to float32 and then resize)
assert numpy.allclose(output_float32.scores, output_uint8.scores, atol=1e-2)

## Manual test for yolact pipeline
pipeline = Pipeline.create(task="yolact")
output_uint8 = pipeline(images=input_uint8)
output_float32 = pipeline(images=input_float32)
assert numpy.allclose(output_float32[0].scores, output_uint8[0].scores, atol=1e-1)
```

### CODE REVIEW

File: src/deepsparse/yolact/utils/utils.py [hunk 1]
```diff
@@ -48,21 +48,25 @@ def preprocess_array(
     :return: preprocessed numpy array (B, C, D, D); where (D,D) is image size expected
         by the network. It is a contiguous array with RGB channel order.
     """
-    image = image.astype(numpy.float32)
+
+    uint8_image = image.dtype == numpy.uint8
```
Comment (rahul-tuli) [inline src/deepsparse/yolact/utils/utils.py:52]: nit: rename to `is_uint8_image`?

**Review by rahul-tuli — DISMISSED** [review]
Super clean!

File: src/deepsparse/yolact/utils/utils.py (hunk 1 above)
Comment (dbogunowicz) [reply src/deepsparse/yolact/utils/utils.py:52]: good idea!

**Review by bfineran — DISMISSED** [review]

**Review by rahul-tuli — DISMISSED** [review]

**Review by bfineran — APPROVED** [review]
Lgtm update pr title to remove openpifpaf paf

**Review by rahul-tuli — APPROVED** [review]

**Pull request merged**